src/data_request_spec.rs

```diff
@@ -1,49 +1,155 @@
 use anyhow::{Context, Result};
+
+use polars::{
+    frame::DataFrame,
+    lazy::{
+        dsl::{col, lit},
+        frame::IntoLazy,
+    },
+    prelude::NamedFrom,
+    series::Series,
+};
 use serde::{Deserialize, Serialize};
 use std::{
     ops::{Index, IndexMut},
     str::FromStr,
 };
 
-use crate::{metadata::SourceDataRelease, parquet::MetricRequest};
+use crate::{
+    metadata::{Metadata, MetricId},
+    parquet::MetricRequest,
+};
 
 #[derive(Serialize, Deserialize, Debug)]
 pub struct DataRequestSpec {
+    pub geometry: GeometrySpec,
     pub region: Vec<RegionSpec>,
     pub metrics: Vec<MetricSpec>,
+    pub years: Option<Vec<String>>,
+}
+
+/// This is the respone to requesting a set of metrics
+/// it contains the matched Metric Requests along with
+/// the geometry that was selected we can also potentially
+/// use this to give feedback on other options in the future
```

Comment:
Split doc comment from comment and add as an enhancement
```suggestion
/// the geometry that was selected.
// TODO (enhancement): We can also potentially use this to give feedback on other options in the future
```